python310Packages.jupyterlab: 3.6.3 -> 4.0.3

dotlambda opened the pull request:
###### Description of changes
Changelog: https://github.com/jupyterlab/jupyterlab/blob/v4.0.3/CHANGELOG.md

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->
closes https://github.com/NixOS/nixpkgs/pull/223814
fixes https://github.com/NixOS/nixpkgs/issues/246725

dotlambda committed:
python310Packages.jupyter-ydoc: 0.3.4 -> 1.0.2

Changelog: https://github.com/jupyter-server/jupyter_ydoc/blob/v1.0.2/CHANGELOG.md

dotlambda committed:
python310Packages.jupyter-collaboration: init at 1.0.1

dotlambda committed:
python310Packages.jupyterlab: 3.6.3 -> 4.0.3

Changelog: https://github.com/jupyterlab/jupyterlab/blob/v4.0.3/CHANGELOG.md

dotlambda committed:
python310Packages.jupyter-server: 2.0.6 -> 2.7.0

Changelog: https://github.com/jupyter-server/jupyter_server/blob/v2.7.0/CHANGELOG.md

nim65s reviewed (changes requested):
`nixpkgs-review pr 244209` is showing:
```
error: builder for '/nix/store/xw2bzyg84g7jw05gx1cwdp0p6p8ybbdd-python3.10-jupyterlab-lsp-4.2.0.drv' failed with exit code 1;
       last 10 log lines:
       > adding 'jupyterlab_lsp-4.2.0.dist-info/RECORD'
       > removing build/bdist.linux-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/jupyterlab-lsp-4.2.0/dist /build/jupyterlab-lsp-4.2.0
       > Processing ./jupyterlab_lsp-4.2.0-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement jupyterlab<4.0.0a0,>=3.6.0 (from jupyterlab-lsp) (from versions: none)
       > ERROR: No matching distribution found for jupyterlab<4.0.0a0,>=3.6.0
       >
```

edit: jupyterlab-lsp is not yet compatible with 4.0.0: https://github.com/jupyter-lsp/jupyterlab-lsp/blob/master/python_packages/jupyterlab_lsp/setup.cfg#L40, ref https://github.com/jupyter-lsp/jupyterlab-lsp/pull/949

dotlambda commented:
> jupyterlab-lsp is not yet compatible with 4.0.0

If that's the only failure I think we should merge anyway.
No other package depends on jupyterlab-lsp. Should it maybe use `buildPythonApplication` and be removed from `python3Packages`?

nim65s commented:
This was the first failure, so I stopped here. I'm running it again.

nim65s commented:
tensorflow 2.11.1 is also failing. 2.13.0 is out, so maybe this can just be bumped. 
I'll let nixpkgs-review run and post the final result.

nim65s commented:
Result of `nixpkgs-review pr 244209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>40 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.batchspawner.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.dockerspawner.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-systemdspawner.dist</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python310Packages.jupyterhub.dist</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.oauthenticator.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.batchspawner.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.dockerspawner.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-systemdspawner.dist</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python311Packages.jupyterhub.dist</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.oauthenticator.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab-lsp.dist</li>
    <li>python311Packages.jupyterlab-lsp</li>
    <li>python311Packages.jupyterlab-lsp.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
  </ul>
</details>
<details>
  <summary>165 packages built:</summary>
  <ul>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>ihaskell</li>
    <li>jupyter</li>
    <li>jupyter.dist</li>
    <li>labplot</li>
    <li>manim</li>
    <li>manim.dist</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.experiment-utilities</li>
    <li>python310Packages.experiment-utilities.dist</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-collaboration</li>
    <li>python310Packages.jupyter-collaboration.dist</li>
    <li>python310Packages.jupyter-contrib-core</li>
    <li>python310Packages.jupyter-contrib-core.dist</li>
    <li>python310Packages.jupyter-contrib-nbextensions</li>
    <li>python310Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-lsp.dist</li>
    <li>python310Packages.jupyter-nbextensions-configurator</li>
    <li>python310Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python310Packages.jupyter-server</li>
    <li>python310Packages.jupyter-server-fileid</li>
    <li>python310Packages.jupyter-server-fileid.dist</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-server-mathjax.dist</li>
    <li>python310Packages.jupyter-server.dist</li>
    <li>python310Packages.jupyter-ydoc</li>
    <li>python310Packages.jupyter-ydoc.dist</li>
    <li>python310Packages.jupyter.dist</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-git.dist</li>
    <li>python310Packages.jupyterlab.dist</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_launcher.dist</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupyterlab_server.dist</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.jupytext.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.layoutparser</li>
    <li>python310Packages.layoutparser.dist</li>
    <li>python310Packages.mkdocs-jupyter</li>
    <li>python310Packages.mkdocs-jupyter.dist</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbclassic.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbdime.dist</li>
    <li>python310Packages.nikola</li>
    <li>python310Packages.nikola.dist</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.nix-kernel.dist</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notebook-shim</li>
    <li>python310Packages.notebook-shim.dist</li>
    <li>python310Packages.notebook.dist</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.notedown.dist</li>
    <li>python310Packages.pdfplumber</li>
    <li>python310Packages.pdfplumber.dist</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.phik.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pymoo</li>
    <li>python310Packages.pymoo.dist</li>
    <li>python310Packages.qgrid</li>
    <li>python310Packages.qgrid.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python310Packages.vector</li>
    <li>python310Packages.vector.dist</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.vega.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.experiment-utilities</li>
    <li>python311Packages.experiment-utilities.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter-collaboration</li>
    <li>python311Packages.jupyter-collaboration.dist</li>
    <li>python311Packages.jupyter-contrib-core</li>
    <li>python311Packages.jupyter-contrib-core.dist</li>
    <li>python311Packages.jupyter-contrib-nbextensions</li>
    <li>python311Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python311Packages.jupyter-lsp</li>
    <li>python311Packages.jupyter-lsp.dist</li>
    <li>python311Packages.jupyter-nbextensions-configurator</li>
    <li>python311Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python311Packages.jupyter-server</li>
    <li>python311Packages.jupyter-server-fileid</li>
    <li>python311Packages.jupyter-server-fileid.dist</li>
    <li>python311Packages.jupyter-server-mathjax</li>
    <li>python311Packages.jupyter-server-mathjax.dist</li>
    <li>python311Packages.jupyter-server.dist</li>
    <li>python311Packages.jupyter-ydoc</li>
    <li>python311Packages.jupyter-ydoc.dist</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.jupyterlab</li>
    <li>python311Packages.jupyterlab-git</li>
    <li>python311Packages.jupyterlab-git.dist</li>
    <li>python311Packages.jupyterlab.dist</li>
    <li>python311Packages.jupyterlab_launcher</li>
    <li>python311Packages.jupyterlab_launcher.dist</li>
    <li>python311Packages.jupyterlab_server</li>
    <li>python311Packages.jupyterlab_server.dist</li>
    <li>python311Packages.jupytext</li>
    <li>python311Packages.jupytext.dist</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.layoutparser.dist</li>
    <li>python311Packages.mkdocs-jupyter</li>
    <li>python311Packages.mkdocs-jupyter.dist</li>
    <li>python311Packages.nbclassic</li>
    <li>python311Packages.nbclassic.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.nbdime</li>
    <li>python311Packages.nbdime.dist</li>
    <li>python311Packages.nikola</li>
    <li>python311Packages.nikola.dist</li>
    <li>python311Packages.nix-kernel</li>
    <li>python311Packages.nix-kernel.dist</li>
    <li>python311Packages.notebook</li>
    <li>python311Packages.notebook-shim</li>
    <li>python311Packages.notebook-shim.dist</li>
    <li>python311Packages.notebook.dist</li>
    <li>python311Packages.notedown</li>
    <li>python311Packages.notedown.dist</li>
    <li>python311Packages.pdfplumber</li>
    <li>python311Packages.pdfplumber.dist</li>
    <li>python311Packages.qgrid</li>
    <li>python311Packages.qgrid.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>quarto</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>wolfram-notebook</li>
  </ul>
</details>

nim65s commented:
Ok, I might not be interpreting nixpkgs-review output correctly, but anyway here is what I got about tensorflow:
```cpp
ERROR: /build/source/tensorflow/python/eager/BUILD:18:11: Compiling tensorflow/python/eager/pywrap_tfe_src.cc failed: (Exit 1): gcc failed: error executing command /nix/store/bxic6j2whyg3z4h2x>
In file included from tensorflow/python/eager/pywrap_tfe_src.cc:55:
./tensorflow/python/util/stack_trace.h: In static member function 'static tensorflow::StackTrace tensorflow::StackTrace::Capture(int)':
./tensorflow/python/util/stack_trace.h:64:55: error: 'PyThreadState' {aka 'struct _ts'} has no member named 'frame'; did you mean 'cframe'?
   64 |     const PyFrameObject* frame = PyThreadState_GET()->frame;
      |                                                       ^~~~~
      |                                                       cframe
./tensorflow/python/util/stack_trace.h:66:56: error: invalid use of incomplete type 'const PyFrameObject' {aka 'const struct _frame'}
   66 |     for (; i < limit && frame != nullptr; frame = frame->f_back, ++i) {
      |                                                        ^~
In file included from bazel-out/k8-opt/bin/external/local_config_python/python_include/Python.h:42,
                 from ./tensorflow/python/lib/core/numpy.h:35,
                 from ./tensorflow/python/eager/pywrap_tensor.h:21,
                 from tensorflow/python/eager/pywrap_tfe_src.cc:51:
bazel-out/k8-opt/bin/external/local_config_python/python_include/pytypedefs.h:22:16: note: forward declaration of 'PyFrameObject' {aka 'struct _frame'}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
./tensorflow/python/util/stack_trace.h:67:37: error: invalid use of incomplete type 'const PyFrameObject' {aka 'const struct _frame'}
   67 |       PyCodeObject* code_obj = frame->f_code;
      |                                     ^~
bazel-out/k8-opt/bin/external/local_config_python/python_include/pytypedefs.h:22:16: note: forward declaration of 'PyFrameObject' {aka 'struct _frame'}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unknown-warning' may have been intended to silence earlier diagnostics

dotlambda commented:
> Ok, I might not be interpreting nixpkgs-review output correctly, but anyway here is what I got about tensorflow

Probably tensorflow isn't rebuilt by this PR. But k-diffusion is and fails because tensorflow does.

dotlambda commented:
jupyter-server fails to build on aarch64-linux and aarch64-darwin but that's probably an issue with upstream

natsukium reviewed (commented):
about jupyter-server:
We can disable `test_regression_is_hidden` for darwin and `test_copy_big_dir` for aarch64-linux.
I built with `doCheck = false` and ran pytest on my local `jupyter_server/` to check the results.

about jupyterlab-lsp:
I don't think jupyterlab-lsp will work with buildPythonApplication, so it will be broken this time.
Migration seems to be in progress.
https://github.com/jupyter-lsp/jupyterlab-lsp/pull/949

Others:
LGTM

dotlambda committed:
python310Packages.notebook: 6.5.2 -> 7.0.1

Changelog: https://github.com/jupyter/notebook/blob/v7.0.1/CHANGELOG.md

dotlambda committed:
python310Packages.jupyterlab_server: 2.19.0 -> 2.24.0

Changelog: https://github.com/jupyterlab/jupyterlab_server/blob/v2.24.0/CHANGELOG.md

dotlambda commented on pkgs/development/python-modules/jupyterlab/default.nix:
```diff
@@ -68,10 +64,10 @@ buildPythonPackage rec {
-    license = with licenses; [ bsd3 ];
+    license = licenses.bsd3;
     homepage = "https://jupyter.org/";
-    maintainers = with maintainers; [ zimbatm ];
```
@zimbatm Would you like to be a member of `teams.jupyter`?

dotlambda commented on pkgs/development/python-modules/notebook/default.nix:
```diff
@@ -3,97 +3,66 @@
+    description = "Web-based notebook environment for interactive computing";
     homepage = "https://github.com/jupyter/notebook";
     license = lib.licenses.bsd3;
-    maintainers = with lib.maintainers; [ fridh ];
```
@FRidh Would you like to be a member of `teams.jupyter`?

nim65s commented:
Result of `nixpkgs-review pr 244209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>42 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.batchspawner.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.dockerspawner.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-systemdspawner.dist</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python310Packages.jupyterhub.dist</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.oauthenticator.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.batchspawner.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.dockerspawner.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-systemdspawner.dist</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python311Packages.jupyterhub.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.oauthenticator.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>16 packages failed to build:</summary>
  <ul>
    <li>python310Packages.experiment-utilities</li>
    <li>python310Packages.experiment-utilities.dist</li>
    <li>python310Packages.jupyter-contrib-nbextensions</li>
    <li>python310Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python310Packages.jupyter-nbextensions-configurator</li>
    <li>python310Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab-lsp.dist</li>
    <li>python311Packages.experiment-utilities</li>
    <li>python311Packages.experiment-utilities.dist</li>
    <li>python311Packages.jupyter-contrib-nbextensions</li>
    <li>python311Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python311Packages.jupyter-nbextensions-configurator</li>
    <li>python311Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python311Packages.jupyterlab-lsp</li>
    <li>python311Packages.jupyterlab-lsp.dist</li>
  </ul>
</details>
<details>
  <summary>153 packages built:</summary>
  <ul>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>ihaskell</li>
    <li>jupyter</li>
    <li>jupyter.dist</li>
    <li>labplot</li>
    <li>manim</li>
    <li>manim.dist</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-collaboration</li>
    <li>python310Packages.jupyter-collaboration.dist</li>
    <li>python310Packages.jupyter-contrib-core</li>
    <li>python310Packages.jupyter-contrib-core.dist</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-lsp.dist</li>
    <li>python310Packages.jupyter-server</li>
    <li>python310Packages.jupyter-server-fileid</li>
    <li>python310Packages.jupyter-server-fileid.dist</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-server-mathjax.dist</li>
    <li>python310Packages.jupyter-server.dist</li>
    <li>python310Packages.jupyter-ydoc</li>
    <li>python310Packages.jupyter-ydoc.dist</li>
    <li>python310Packages.jupyter.dist</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-git.dist</li>
    <li>python310Packages.jupyterlab.dist</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_launcher.dist</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupyterlab_server.dist</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.jupytext.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.layoutparser</li>
    <li>python310Packages.layoutparser.dist</li>
    <li>python310Packages.mkdocs-jupyter</li>
    <li>python310Packages.mkdocs-jupyter.dist</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbclassic.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbdime.dist</li>
    <li>python310Packages.nikola</li>
    <li>python310Packages.nikola.dist</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.nix-kernel.dist</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notebook-shim</li>
    <li>python310Packages.notebook-shim.dist</li>
    <li>python310Packages.notebook.dist</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.notedown.dist</li>
    <li>python310Packages.pdfplumber</li>
    <li>python310Packages.pdfplumber.dist</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.phik.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pymoo</li>
    <li>python310Packages.pymoo.dist</li>
    <li>python310Packages.qgrid</li>
    <li>python310Packages.qgrid.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python310Packages.vector</li>
    <li>python310Packages.vector.dist</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.vega.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter-collaboration</li>
    <li>python311Packages.jupyter-collaboration.dist</li>
    <li>python311Packages.jupyter-contrib-core</li>
    <li>python311Packages.jupyter-contrib-core.dist</li>
    <li>python311Packages.jupyter-lsp</li>
    <li>python311Packages.jupyter-lsp.dist</li>
    <li>python311Packages.jupyter-server</li>
    <li>python311Packages.jupyter-server-fileid</li>
    <li>python311Packages.jupyter-server-fileid.dist</li>
    <li>python311Packages.jupyter-server-mathjax</li>
    <li>python311Packages.jupyter-server-mathjax.dist</li>
    <li>python311Packages.jupyter-server.dist</li>
    <li>python311Packages.jupyter-ydoc</li>
    <li>python311Packages.jupyter-ydoc.dist</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.jupyterlab</li>
    <li>python311Packages.jupyterlab-git</li>
    <li>python311Packages.jupyterlab-git.dist</li>
    <li>python311Packages.jupyterlab.dist</li>
    <li>python311Packages.jupyterlab_launcher</li>
    <li>python311Packages.jupyterlab_launcher.dist</li>
    <li>python311Packages.jupyterlab_server</li>
    <li>python311Packages.jupyterlab_server.dist</li>
    <li>python311Packages.jupytext</li>
    <li>python311Packages.jupytext.dist</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.layoutparser.dist</li>
    <li>python311Packages.mkdocs-jupyter</li>
    <li>python311Packages.mkdocs-jupyter.dist</li>
    <li>python311Packages.nbclassic</li>
    <li>python311Packages.nbclassic.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.nbdime</li>
    <li>python311Packages.nbdime.dist</li>
    <li>python311Packages.nikola</li>
    <li>python311Packages.nikola.dist</li>
    <li>python311Packages.nix-kernel</li>
    <li>python311Packages.nix-kernel.dist</li>
    <li>python311Packages.notebook</li>
    <li>python311Packages.notebook-shim</li>
    <li>python311Packages.notebook-shim.dist</li>
    <li>python311Packages.notebook.dist</li>
    <li>python311Packages.notedown</li>
    <li>python311Packages.notedown.dist</li>
    <li>python311Packages.pdfplumber</li>
    <li>python311Packages.pdfplumber.dist</li>
    <li>python311Packages.qgrid</li>
    <li>python311Packages.qgrid.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>quarto</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>wolfram-notebook</li>
  </ul>
</details>

natsukium commented:
I'm not sure why these packages had been successfully built.
python310Packages.experiment-utilities
python310Packages.jupyter-contrib-nbextensions
python310Packages.jupyter-nbextensions-configurator

Should we patch them to use `nbclassic` or `jupyter-server` instead of `notebook>=7`?

natsukium commented:
Anyway, I think it should be reported upstream.

dotlambda committed:
python310Packages.ipynbname: init at 2023.2.0.0

dotlambda committed:
python310Packages.experiment-utilities: unvendor ipynbname

dotlambda commented on pkgs/development/python-modules/experiment-utilities/unvendor-ipynbname.patch:
@ChrisReinke Please consider applying something like this patch upstream. I don't have an account on gitlab.inria.fr otherwise I would make a merge request.

dotlambda commented:
> Anyway, I think it should be reported upstream.

* https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1647
* https://github.com/Jupyter-contrib/jupyter_nbextensions_configurator/issues/165

dotlambda committed:
python310Packages.jupyter-nbextensions-configurator: make compatible with notebook v7

dotlambda commented:
> Should we patch them to use `nbclassic` or `jupyter-server` instead of `notebook>=7`?

Done. But I did not test their functionality.

natsukium commented:
Thanks! LGTM for x86_64-linux.

It seems to fail on aarch64-linux, I will check it later.

dotlambda commented:
Result of `nixpkgs-review pr 244209` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>54 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.batchspawner.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.dockerspawner.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.encodec</li>
    <li>python310Packages.encodec.dist</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-systemdspawner.dist</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python310Packages.jupyterhub.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.oauthenticator.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.batchspawner.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.dockerspawner.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-systemdspawner.dist</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python311Packages.jupyterhub.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.oauthenticator.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>labplot</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-kernel.dist</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab-lsp.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.jupyterlab-lsp</li>
    <li>python311Packages.jupyterlab-lsp.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
  </ul>
</details>
<details>
  <summary>142 packages built:</summary>
  <ul>
    <li>jupyter</li>
    <li>jupyter.dist</li>
    <li>manim</li>
    <li>manim.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.experiment-utilities</li>
    <li>python310Packages.experiment-utilities.dist</li>
    <li>python310Packages.ipynbname</li>
    <li>python310Packages.ipynbname.dist</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-collaboration</li>
    <li>python310Packages.jupyter-collaboration.dist</li>
    <li>python310Packages.jupyter-contrib-core</li>
    <li>python310Packages.jupyter-contrib-core.dist</li>
    <li>python310Packages.jupyter-contrib-nbextensions</li>
    <li>python310Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-lsp.dist</li>
    <li>python310Packages.jupyter-nbextensions-configurator</li>
    <li>python310Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python310Packages.jupyter-server</li>
    <li>python310Packages.jupyter-server-fileid</li>
    <li>python310Packages.jupyter-server-fileid.dist</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-server-mathjax.dist</li>
    <li>python310Packages.jupyter-server.dist</li>
    <li>python310Packages.jupyter-ydoc</li>
    <li>python310Packages.jupyter-ydoc.dist</li>
    <li>python310Packages.jupyter.dist</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-git.dist</li>
    <li>python310Packages.jupyterlab.dist</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_launcher.dist</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupyterlab_server.dist</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.jupytext.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.layoutparser</li>
    <li>python310Packages.layoutparser.dist</li>
    <li>python310Packages.mkdocs-jupyter</li>
    <li>python310Packages.mkdocs-jupyter.dist</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbclassic.dist</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbdime.dist</li>
    <li>python310Packages.nikola</li>
    <li>python310Packages.nikola.dist</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.nix-kernel.dist</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notebook-shim</li>
    <li>python310Packages.notebook-shim.dist</li>
    <li>python310Packages.notebook.dist</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.notedown.dist</li>
    <li>python310Packages.pdfplumber</li>
    <li>python310Packages.pdfplumber.dist</li>
    <li>python310Packages.phik</li>
    <li>python310Packages.phik.dist</li>
    <li>python310Packages.pymoo</li>
    <li>python310Packages.pymoo.dist</li>
    <li>python310Packages.qgrid</li>
    <li>python310Packages.qgrid.dist</li>
    <li>python310Packages.vector</li>
    <li>python310Packages.vector.dist</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.vega.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.experiment-utilities</li>
    <li>python311Packages.experiment-utilities.dist</li>
    <li>python311Packages.ipynbname</li>
    <li>python311Packages.ipynbname.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter-collaboration</li>
    <li>python311Packages.jupyter-collaboration.dist</li>
    <li>python311Packages.jupyter-contrib-core</li>
    <li>python311Packages.jupyter-contrib-core.dist</li>
    <li>python311Packages.jupyter-contrib-nbextensions</li>
    <li>python311Packages.jupyter-contrib-nbextensions.dist</li>
    <li>python311Packages.jupyter-lsp</li>
    <li>python311Packages.jupyter-lsp.dist</li>
    <li>python311Packages.jupyter-nbextensions-configurator</li>
    <li>python311Packages.jupyter-nbextensions-configurator.dist</li>
    <li>python311Packages.jupyter-server</li>
    <li>python311Packages.jupyter-server-fileid</li>
    <li>python311Packages.jupyter-server-fileid.dist</li>
    <li>python311Packages.jupyter-server-mathjax</li>
    <li>python311Packages.jupyter-server-mathjax.dist</li>
    <li>python311Packages.jupyter-server.dist</li>
    <li>python311Packages.jupyter-ydoc</li>
    <li>python311Packages.jupyter-ydoc.dist</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.jupyterlab</li>
    <li>python311Packages.jupyterlab-git</li>
    <li>python311Packages.jupyterlab-git.dist</li>
    <li>python311Packages.jupyterlab.dist</li>
    <li>python311Packages.jupyterlab_launcher</li>
    <li>python311Packages.jupyterlab_launcher.dist</li>
    <li>python311Packages.jupyterlab_server</li>
    <li>python311Packages.jupyterlab_server.dist</li>
    <li>python311Packages.jupytext</li>
    <li>python311Packages.jupytext.dist</li>
    <li>python311Packages.layoutparser</li>
    <li>python311Packages.layoutparser.dist</li>
    <li>python311Packages.mkdocs-jupyter</li>
    <li>python311Packages.mkdocs-jupyter.dist</li>
    <li>python311Packages.nbclassic</li>
    <li>python311Packages.nbclassic.dist</li>
    <li>python311Packages.nbdime</li>
    <li>python311Packages.nbdime.dist</li>
    <li>python311Packages.nikola</li>
    <li>python311Packages.nikola.dist</li>
    <li>python311Packages.nix-kernel</li>
    <li>python311Packages.nix-kernel.dist</li>
    <li>python311Packages.notebook</li>
    <li>python311Packages.notebook-shim</li>
    <li>python311Packages.notebook-shim.dist</li>
    <li>python311Packages.notebook.dist</li>
    <li>python311Packages.notedown</li>
    <li>python311Packages.notedown.dist</li>
    <li>python311Packages.pdfplumber</li>
    <li>python311Packages.pdfplumber.dist</li>
    <li>python311Packages.qgrid</li>
    <li>python311Packages.qgrid.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>sage</li>
    <li>sageWithDoc</li>
  </ul>
</details>
